### DISCUSSION TITLE
OM development environment setup

### DISCUSSION THREAD

**Discussion by lijunliangTG** [OpenModelica/OpenModelica#10374] [Q&A]
1、When you complete the installation steps, you will find that qt does not contain OM's webkit library, what version of qt did you use to develop?

2、Is there any development manual that can be shared so that developers can get started as soon as possible?

**Comment by mahge** [comment]
Hi @lijunliangTG. I think we need little more information. What Operating System are you using? Which installation steps did you follow?

**Comment by lijunliangTG** [comment]
windows10  https://github.com/OpenModelica/OpenModelica/blob/master/OMCompiler/README.Windows.md this steps

**Comment by adeas31** [comment]
Qt is part of OMDev. Read the OMDev `INSTALL.txt` and `README.txt` to properly setup the development environment for Windows.

**Comment by lijunliangTG** [comment]
now I finish build modelica  ,here is an another problem Project ERROR: Unknown module(s) in QT: webkit webkitwidgets ,how  can I solve them

**Comment by mahge** [comment]
@lijunliangTG You have to give us as much information as possible. Please take the time and document what you did and what errors messages you got exactly. Otherwise it is difficult for us to give any suggestions or fixes.

**Comment by lijunliangTG** [comment]
when i use this command 
make -f Makefile.omdev.mingw -j 4 omedit
i meet the problem below

```
 Run Build Command(s):C:/Program Files/CMake/bin/cmake.exe -E env VERBOSE=1 C:/OMDev/tools/msys/usr/bin/make.exe -f Makefile cmTC_2a76f/fast && make[3]: 进入目录“/e/MyOM/OpenModelica/OMParser/3rdParty/antlr4/runtime/Cpp/build/CMakeFiles/CMakeScratch/TryCompile-3v98zz”
    /usr/bin/make  -f CMakeFiles/cmTC_2a76f.dir/build.make CMakeFiles/cmTC_2a76f.dir/build
    make[4]: 进入目录“/e/MyOM/OpenModelica/OMParser/3rdParty/antlr4/runtime/Cpp/build/CMakeFiles/CMakeScratch/TryCompile-3v98zz”
    Building C object CMakeFiles/cmTC_2a76f.dir/testCCompiler.c.obj
    "C:/Program Files/LLVM/bin/clang.exe"   -O0 -g -Xclang -gcodeview -D_DEBUG -D_DLL -D_MT -Xclang --dependent-lib=msvcrtd -MD -MT CMakeFiles/cmTC_2a76f.dir/testCCompiler.c.obj -MF CMakeFiles/cmTC_2a76f.dir/testCCompiler.c.obj.d -o CMakeFiles/cmTC_2a76f.dir/testCCompiler.c.obj -c E:/MyOM/OpenModelica/OMParser/3rdParty/antlr4/runtime/Cpp/build/CMakeFiles/CMakeScratch/TryCompile-3v98zz/testCCompiler.c
    Linking C executable cmTC_2a76f.exe
    "C:/Program Files/LLVM/bin/clang.exe" -fuse-ld=lld-link -nostartfiles -nostdlib -O0 -g -Xclang -gcodeview -D_DEBUG -D_DLL -D_MT -Xclang --dependent-lib=msvcrtd -Xlinker /subsystem:console @CMakeFiles/cmTC_2a76f.dir/objects1.rsp -o cmTC_2a76f.exe -Xlinker /MANIFEST:EMBED -Xlinker /implib:cmTC_2a76f.lib -Xlinker /pdb:E:/MyOM/OpenModelica/OMParser/3rdParty/antlr4/runtime/Cpp/build/CMakeFiles/CMakeScratch/TryCompile-3v98zz/cmTC_2a76f.pdb -Xlinker /version:0.0  @CMakeFiles/cmTC_2a76f.dir/linkLibs.rsp
    lld-link: error: could not open 'kernel32.lib': no such file or directory
    lld-link: error: could not open 'user32.lib': no such file or directory
    lld-link: error: could not open 'gdi32.lib': no such file or directory
    lld-link: error: could not open 'winspool.lib': no such file or directory
    lld-link: error: could not open 'shell32.lib': no such file or directory
    lld-link: error: could not open 'ole32.lib': no such file or directory
    lld-link: error: could not open 'oleaut32.lib': no such file or directory
    lld-link: error: could not open 'uuid.lib': no such file or directory
    lld-link: error: could not open 'comdlg32.lib': no such file or directory
    lld-link: error: could not open 'advapi32.lib': no such file or directory
    lld-link: error: could not open 'oldnames.lib': no such file or directory
    lld-link: error: could not open 'msvcrtd.lib': no such file or directory
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    make[4]: *** [CMakeFiles/cmTC_2a76f.dir/build.make:101：cmTC_2a76f.exe] 错误 1
    make[4]: 离开目录“/e/MyOM/OpenModelica/OMParser/3rdParty/antlr4/runtime/Cpp/build/CMakeFiles/CMakeScratch/TryCompile-3v98zz”
    make[3]: *** [Makefile:127：cmTC_2a76f/fast] 错误 2
    make[3]: 离开目录“/e/MyOM/OpenModelica/OMParser/3rdParty/antlr4/runtime/Cpp/build/CMakeFiles/CMakeScratch/TryCompile-3v98zz”





  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:26 (project)


-- Configuring incomplete, errors occurred!
make[2]: *** [Makefile:34：3rdParty/antlr4/runtime/Cpp/build/Makefile] 错误 1
make[2]: 离开目录“/e/MyOM/OpenModelica/OMParser”
make[1]: *** [Makefile:38：install/lib/libantlr4-runtime.a] 错误 2
make[1]: 离开目录“/e/MyOM/OpenModelica/OMParser”
make: *** [Makefile.omdev.mingw:99：omparser] 错误 2


```

**Comment by mahge** [comment]
> "C:/Program Files/LLVM/bin/clang.exe"

This suggests that you are trying to build OpenModelica using the Windows Command Prompt. That will not work.

Please follow these instructions https://github.com/OpenModelica/OpenModelica/blob/master/OMCompiler/README.Windows.md#1-omdev-package
up until section 1.4 to get `OMDev `properly setup. 

From now on make sure you are using a MSYS shell when working with OpenModelica. You can start it by going to `<your_OMDev_directory>\tools\msys\mingw64.exe`. You need to use this shell to build OpenModelica. 

Once you have opened the MSYS shell, you should clean your OpenModelica repository to avoid issues from your previous attempts. NOTE that this will delete any new files you have added into the OpenModelica directory.

```
cd /path/to/OpenModelica
git clean -fxd 
git submodule foreach --recursive git clean -fxd
```

Now you can build OpenModelica as follows:

```
cd /path/to/OpenModelica
cmake -S . -B build_cmake -Wno-dev -G "MSYS Makefiles"
cd build_cmake
# Replace <Nr. of cores> with the number of your cores
make -j<Nr. of cores> install -Oline
```

**Comment by lijunliangTG** [comment]
thanks for you patient answer . when i execute `cmake -S . -B build_cmake -Wno-dev -G "MSYS Makefiles" `  ,i stil meet the CMake  problem below 


[CMakeError.log](https://github.com/OpenModelica/OpenModelica/files/10926339/CMakeError.log)
[CMakeOutput.log](https://github.com/OpenModelica/OpenModelica/files/10926341/CMakeOutput.log)

**Comment by mahge** [comment]
Thank you. Can you also attach the actual output of CMake. I mean what CMake printed to the terminal. There is still something not quite right here.

**Comment by lijunliangTG** [comment]
I will attach the CMake printed to the terminal  in the below file
[CMAKE_PRINT.txt](https://github.com/OpenModelica/OpenModelica/files/10928976/CMAKE_PRINT.txt)

**Comment by mahge** [comment]
Okay you just do not have the submodules of OpenModelica. The OpenModelica repository has some submodules. You can run this command to initialize them
```sh
cd /path/to/OpenModelica
git submodule update --init --recursive
```

The try compiling again. 

-------------------------------------------------------------------------------------------
In the future you can clone the repository with the submodules using 
```sh
# note the recursive flag at the end.
git clone https://github.com/OpenModelica/OpenModelica.git --recursive 
```

**Reply by lijunliangTG** [reply]
when I using `make -j 6 install -Oline` some error occured. Is this still some files missing ?How could I solve them

[make comend error.txt](https://github.com/OpenModelica/OpenModelica/files/10937548/make.error.txt)
[CMakeError.log](https://github.com/OpenModelica/OpenModelica/files/10937561/CMakeError.log)
[CMakeOutput.log](https://github.com/OpenModelica/OpenModelica/files/10937562/CMakeOutput.log)

**Reply by mahge** [reply]
I somehow missed the notification for your comment. Sorry.

First make sure you have set the environment variable OMDEV to point to your OMDev location as described [here](https://github.com/OpenModelica/OpenModelica/blob/master/OMCompiler/README.Windows.md#12-install-omdev). 

If that doesn't work please delete your OMDev and restart the whole process following the [instructions](https://github.com/OpenModelica/OpenModelica/blob/master/OMCompiler/README.Windows.md#1-omdev-package) step by step.

**Reply by lijunliangTG** [reply]
I have  re-try the step ,but when do  `cmake -S . -B build_cmake -Wno-dev -G "MSYS Makefiles" ` these files doesn't found 

![image](https://user-images.githubusercontent.com/117806079/224525858-4b11a393-9ba5-4506-ad8a-0bd92cc630bf.png)

Is it because of the absence of these files that caused my compilation to fail?
